src/clm/functions.py

```diff
@@ -143,13 +143,22 @@ def read_file(smiles_file, max_lines=None):
     return lines_array
 
 
-def read_file_incremental(input_file, max_lines=None):
+def read_file_incremental(
+    input_file,
+    max_lines=None,
+    smile_only=True,
+):
     count = 0
-    for line in open(input_file, "r").readlines():
-        yield line.strip()
-        count += 1
-        if max_lines is not None and count == max_lines:
-            return
+    with open(input_file, "r") as f:
+        # Extract index of smiles from header
+        if not smile_only:
```

Comment:
`smile_only` indicates what the caller wants this function to return, not the format of the file.

Reply:
Got it! Updated the usage of `smile_only` variable accordingly.